wagtail/contrib/wagtailapi/serializers.py

```diff
@@ -93,80 +88,90 @@ def serialize_object_metadata(self, request, obj, show_details=False):
 
         return data
 
-    def serialize_object(self, request, obj, fields=frozenset(), extra_data=(), all_fields=False, show_details=False):
-        """
-        This converts an object into JSON-serialisable dict so it can
-        be used in the API.
-        """
+    def serialize_fields(self, obj):
+        # Call rest frameworks built in to_representation method
+        return super(BaseSerializer, self).to_representation(obj)
+
+    def to_representation(self, obj):
         data = [
             ('id', obj.id),
         ]
 
-        # Add meta
-        metadata = self.serialize_object_metadata(request, obj, show_details=show_details)
+        # Serialize meta
+        metadata = self.serialize_meta(obj)
         if metadata:
             data.append(('meta', metadata))
```

Comment:
Could `meta` and `id` instead be regular fields, rather than overriding `representation`?